ddsrouter_core/src/cpp/writer/implementations/rtps/rpc/RequestWriter.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file RequestWriter.hpp
+ */
+
+#ifndef __SRC_DDSROUTERCORE_WRITER_IMPLEMENTATIONS_RTPS_RPC_REQUESTWRITER_HPP_
+#define __SRC_DDSROUTERCORE_WRITER_IMPLEMENTATIONS_RTPS_RPC_REQUESTWRITER_HPP_
+
+#include <communication/rpc/ServiceRegistry.hpp>
+#include <ddsrouter_core/types/participant/ParticipantId.hpp>
+#include <ddsrouter_core/types/topic/RPCTopic.hpp>
+
+#include <writer/implementations/rtps/Writer.hpp>
+
+namespace eprosima {
+namespace ddsrouter {
+namespace core {
+namespace rtps {
+
+class RequestWriter : public Writer
```

Comment:
For `Reply` and `RequestWirter`. This is very difficult to keep specializing (for example in case of Repeater, where specific Writer is needed). Couldn't this be done by generic intermediate functions?
e.g. in `rtps::Writer` have a virtual method called `set_wparams_` and these two classes only set implement this method, while other Writers has it empty. (just an idea, I have not though of consequences of this)

Reply:
I removed the new Writers and tried to do something as suggested, allowing to only have the write logic in one place.